File: simjaesung/week16/724.java
class Solution {
	public int pivotIndex(int[] nums) {
		int n = nums.length;
		int[] arr = new int[n];
		arr[0] = nums[0];
		for(int i = 1; i < n; i++){
			arr[i] = arr[i - 1] + nums[i];
		}

		for(int i = 0; i < n; i++){
			if(arr[i] - nums[i] == arr[n-1] - arr[i]){
				return i;
			}
		}

		return -1;
	}
}

File: yiseungyun/week16/724.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function(nums) {
  let pivot = 0;
  let left = 0, right = sumOfArray(1, nums.length, nums);
  while (pivot < nums.length) {
      if (left === right) return pivot;
      left += nums[pivot];
      right -= nums[pivot+1];
      pivot++;
  }
  return -1;
};

function sumOfArray(start, end, array) {
  let sum = 0;
  for (let i = start; i < end; i++) {
      sum += array[i];
  }
  return sum;
}